tests/explicit-nulls/neg/flow-match.scala

```diff
@@ -0,0 +1,17 @@
+// Test flow-typing when NotNullInfos are from cases
+
+object MatchTest {
+
+  def stringOrNullToString(s: String | Null): String = s match {
+    case null => "null"
+    // after the null case, s becomes non-nullable
+    case _ => s
+  }
+
+  def stringOrNullToString2(s: String | Null): String = s match {
+    case null => "null"
+    // s in the case is new variable different from the paramter s
```

Comment:
This reasoning doesn't make sense to me, as I stated in the companion issue.
The newly defined `s` can't be `null`, because null is checked by the previous branch.
It's type should be `String` in my opinion.

Reply:
I agree, we should also give the better type to the pattern-bound `s`.

Reply:
Removed